### PR TITLE
tower-abci: support CometBFT `0.38.x`

### PR DESCRIPTION
**Pull request by erwanor**
This PR adds support for `0.38.x` and adds a key-value store over ABCI example for that version.

### CODE REVIEW

**Comment by erwanor** [comment]
We want to get this into a tagged release ahead of deciding whether/when we will move to `cometbft-rs`.

File: examples/kvstore_38/main.rs [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+        response::Query {
+            log,
+            key: key.into_bytes().into(),
+            value: value.into_bytes().into(),
+            ..Default::default()
+        }
+    }
+
+    fn execute_tx(&mut self, tx: Bytes) -> ExecTxResult {
```
Comment (joroshiba) [inline examples/kvstore_38/main.rs:110]: Execute TX feels odd in the examples here since it's a byproduct of `finalize_block` for 0.38?

Comment (erwanor) [reply examples/kvstore_38/main.rs:110]: Would it be clearer if it was inline in the `finalize_block` method?

Comment (erwanor) [reply examples/kvstore_38/main.rs:110]: Ah I think I understand what you mean, to clarify: this is exactly an example for `0.38`. There are others for `v0.37` and `v0.34` respectively, but not in this PR diff.

Comment (joroshiba) [reply examples/kvstore_38/main.rs:110]: Gotcha, I think I understand the purpose better now. Think this is perfectly fine. Was just confused by a non abci++  method in the example here

**Comment by cratelyn** [comment]
thank you for doing this, @erwanor!

**Pull request merged**